apps/calendar/js/calc.js

```diff
@@ -36,22 +36,26 @@ exports.daysInWeek = function() {
 };
 
 /**
- * Calculates day of week when starting day is Monday.
+ * Calculates day of week from startDay value
+ * passed by the locale currently being used
  */
-exports.dayOfWeekFromMonday = function(numeric) {
-  var day = numeric - 1;
+exports.dayOfWeekFromStartDay = function(numeric) {
+  var day = numeric - exports.startDay;
   if (day < 0) {
-    return 6;
+    return 7 + day;
   }
-
   return day;
 };
 
 /**
- * Calculates day of week when starting day is Sunday.
+ * Calculates day of week when starting day is Monday.
  */
-exports.dayOfWeekFromSunday = function(numeric) {
-  return numeric;
+exports.dayOfWeekFromMonday = function(numeric) {
```

Comment:
nope, because on `Date.getDay()` `0 == Sunday` and `startDay` can be `0`, `1` or `6`.. so it would be very confusing.. we really need to get how far the day is from the start of the work week (Monday) on the week view.